### PR TITLE
Add imagePullSecrets property to SPOD configuration

### PR DESCRIPTION
**Pull request by ccojocar**
#### What type of PR is this?

/kind api-change

#### What this PR does / why we need it:

Adds the `imagePullSecrets` property to SPOD configuration. This is required to pull the images from spod pod from a private registry. For instance when these images are mirrored in a private registry instead of using them from public registry.

#### Does this PR introduce a user-facing change?

```release-note
Add imagePullSecrets to SPOD configuration.

```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]

File: internal/pkg/manager/spod/spod_controller.go [hunk 1]
```diff
@@ -587,6 +587,7 @@ func (r *ReconcileSPOd) getConfiguredSPOd(
 
 	templateSpec.Tolerations = cfg.Spec.Tolerations
 	templateSpec.Affinity = cfg.Spec.Affinity
+	templateSpec.ImagePullSecrets = cfg.Spec.ImagePullSecrets
```
Comment (saschagrunert) [inline internal/pkg/manager/spod/spod_controller.go:590]: I'm wondering if we should apply this to all other images we pull as well. :thinking:

**Comment by saschagrunert** [comment]
CI is not happy with the TOC: 
```
+- [Pull images from private registry](#pull-images-from-private-registry)
```

File: internal/pkg/manager/spod/spod_controller.go (hunk 1 above)
Comment (ccojocar) [reply internal/pkg/manager/spod/spod_controller.go:590]: Do you mean in the webhook binding?

Comment (saschagrunert) [reply internal/pkg/manager/spod/spod_controller.go:590]: Yeah for example, we also use kube rbac proxy and things like that but their manifests are mostly static.

Comment (ccojocar) [reply internal/pkg/manager/spod/spod_controller.go:590]: I configured it also in the webhook default deployment. Are there any other place which we should consider?

Comment (saschagrunert) [reply internal/pkg/manager/spod/spod_controller.go:590]: The daemon probably.

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
@jhrozek @JAORMX PTAL
/hold for their eyes

File: internal/pkg/manager/spod/spod_controller.go (hunk 1 above)
Comment (ccojocar) [reply internal/pkg/manager/spod/spod_controller.go:590]: I am a bit confused. Isn't the spod the daemon? I think the other resources get deployed declaratively using manifests.

**Review by jhrozek — APPROVED** [review]
LGTM, too

**Comment by ccojocar** [comment]
Thanks @jhrozek  for review, @saschagrunert Can we merge this now? Thanks

**Review by saschagrunert — APPROVED** [review]
/unhold

**Pull request merged**